src/workshed/index.ts

```diff
@@ -0,0 +1,143 @@
+import { getWorkshed, haveEffect, Item, print, totalTurnsPlayed, use, visitUrl } from "kolmafia";
+import { $effect, $item, $items, AsdonMartin, clamp, DNALab, get, have, TrainSet } from "libram";
+import { dietCompleted } from "../diet";
+import { globalOptions } from "../config";
+import { potionSetupCompleted } from "../potions";
+import { estimatedTurns } from "../turns";
+import { getBestCycle, grabMedicine, offsetDefaultPieces } from "./utils";
+
+let _nextWorkshed: GarboWorkshed | null = null;
+let _currentWorkshed: GarboWorkshed | null = null;
+class GarboWorkshed {
+  workshed: Item;
+  done = () => true;
```

Comment:
it took a little bit to wrap my head around the boolean logic being used for the "done" guy here tbh. it's fine but it did take a bit of thought and isn't super intuitive imho. i'm not sure GarboWorkshed as a class is great at introducing users on how to make GarboWorkshed entries for new ones, which doesn't matter now but might be a little annoying cruft down the road.

Reply:
I should make the constructor take an options object to make it easier to read, and maybe use a more informative name than done